test/parallel/test-net-persistent-nodelay.js

```diff
@@ -0,0 +1,43 @@
+var common = require('../common');
+var assert = require('assert');
+var net = require('net');
+
+var serverConnection;
+var echoServer = net.createServer(function(connection) {
+  serverConnection = connection;
+  connection.end()
+});
+echoServer.listen(common.PORT);
+
+var setTrue = false;
+
+var Socket = net.Socket;
+
+Socket.prototype.setNoDelay = function(enable) {
+  this._setNoDelay = enable === undefined ? true : !!enable;
+  if (this._handle && this._handle.setNoDelay) {
+    if (enable === undefined || !!enable) {
+      setTrue = true;
+    }
+    this._handle.setNoDelay(this._setNoDelay);
+  }
+}
+
+echoServer.on('listening', function() {
+  var sock1 = new Socket({});
```

Comment:
After calling the constructor, you could assert that `_setNoDelay` has the proper default value. You could do something similar in the other test.